extensions/libraries/joomla_entity/src/Fields/Traits/HasFields.php

```diff
@@ -79,50 +97,92 @@ public function fieldValue($id, $default = null, $raw = false)
 			throw new \InvalidArgumentException($msg);
 		}
 
-		return (null === $values[$id]) ? $default : $values[$id];
+		return (null === $values[$id]) ? $default : $values[$id][$raw ? 'rawvalue' : 'value'];
 	}
 
 	/**
-	 * Get all the field values for this entity.
+	 * Retrieve a field value by a given field name
 	 *
-	 * @param   boolean  $raw  Return raw field values
+	 * @param   integer  $name     Name of the field which value we want to retrieve.
+	 * @param   mixed    $default  Value to use as default if value is null
+	 * @param   boolean  $raw      Return raw field value
+	 *
+	 * @return  mixed
+	 *
+	 * @throws  \InvalidArgumentException  If field value is not set
+	 */
+	public function fieldValueByName($name, $default = null, $raw = false)
+	{
+		$values = $this->fieldValues();
+		$id = array_search($name, array_column($values, 'name'));
+
+		if ($id === false)
+		{
+			return $default;
+		}
+
+		return array_values($values)[$id][$raw ? 'rawvalue' : 'value'];
+	}
+
+	/**
+	 * Get all the field values for this entity.
 	 *
 	 * @return  array
 	 */
-	public function fieldValues($raw = false)
+	public function fieldValues()
 	{
-		$values = array();
-		$fields = $this->fields();
+		// Returns the cached array of values
+		if (null !== $this->fieldValues)
```

Comment:
If we do this we need to methods: keep `fieldValues()` clean and move all the logic to `loadFieldValues()`